examples/warp-key-value-store/src/main.rs

```diff
@@ -0,0 +1,200 @@
+use bytes::Bytes;
+use hyper::{
+    body::HttpBody,
+    header::{self, HeaderValue},
+    Body, Request, Response, Server, StatusCode,
+};
+use std::collections::HashMap;
+use std::future::Future;
+use std::net::SocketAddr;
+use std::sync::{Arc, RwLock};
+use std::time::Duration;
+use structopt::StructOpt;
+use tower::{make::Shared, BoxError, Service, ServiceBuilder};
+use tower_http::{
+    add_extension::AddExtensionLayer,
+    compression::{CompressionBody, CompressionLayer},
+    sensitive_header::SetSensitiveHeaderLayer,
+    set_response_header::SetResponseHeaderLayer,
+};
+use warp::{filters, path};
+use warp::{Filter, Rejection, Reply};
+
+/// Simple key/value store with an HTTP API
+#[derive(Debug, StructOpt)]
+struct Config {
+    /// The port to listen on
+    #[structopt(long, short = "p", default_value = "3000")]
+    port: u16,
+}
+
+type Database = Arc<RwLock<HashMap<String, Bytes>>>;
+
+#[tokio::main]
+async fn main() {
+    // Setup tracing
+    tracing_subscriber::fmt::init();
+
+    // Parse command line arguments
+    let config = Config::from_args();
+
+    // Build our service.
+    let service = make_service();
```

Comment:
I think we can have a better name for this function, that describes better what its doing.

Reply:
I found a way to avoid having the service returned by a function. Should be simpler :)